lightning/src/ln/functional_tests.rs

```diff
@@ -10511,6 +10511,90 @@ fn test_remove_expired_inbound_unfunded_channels() {
 	check_closed_event(&nodes[1], 1, ClosureReason::HolderForceClosed, false, &[nodes[0].node.get_our_node_id()], 100000);
 }
 
+#[test]
+	fn test_channel_close_when_not_timely_accepted() {
+		// Create network of two nodes
+		let chanmon_cfgs = create_chanmon_cfgs(2);
+		let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+		let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+		// Simulate peer-diconnects mid-handshake
+		// The channel is initiated from the node 0 side,
+		// But the nodes disconnects before node 1 could send accept channel
```

Comment:
disconnect

Reply:
Fixed in 15792af
Thanks for catching it!